tests/bounded_bls12381.rs

```diff
@@ -0,0 +1,127 @@
+        challengeHash: result.challengeHash.clone(),
+        messageCount: 10,
+        publicKey: dpk0,
+        nonce: b"dummy nonce".to_vec(),
+    };
+    let js_value = serde_wasm_bindgen::to_value(&request).unwrap();
+    let res = verify_bounded_bls_signature_request(js_value).await;
+    assert!(res.is_ok());
+    let res: BbsVerifyResponse = serde_wasm_bindgen::from_value(res.unwrap()).unwrap();
+    assert!(res.verified);
+
+    let request = BoundedBlsSignatureVerifyContextRequest {
+        commitment: result.commitment.clone(),
+        proofOfHiddenMessages: result.proofOfHiddenMessages.clone(),
+        challengeHash: result.challengeHash.clone(),
+        messageCount: 10,
+        publicKey: dpk0,
+        nonce: b"bad nonce".to_vec(),
+    };
+    let js_value = serde_wasm_bindgen::to_value(&request).unwrap();
+    let res = verify_bounded_bls_signature_request(js_value).await;
+    assert!(res.is_ok());
+    let res: BbsVerifyResponse = serde_wasm_bindgen::from_value(res.unwrap()).unwrap();
+    assert!(!res.verified);
+}
+
+#[allow(non_snake_case)]
+#[wasm_bindgen_test]
+pub async fn bounded_bls_sign_tests() {
+    let key_pair_js0 = bls_generate_g2_key(Some(vec![0u8; 16])).await.unwrap();
+    let dpk_bytes0 = serde_wasm_bindgen::from_value::<BlsKeyPair>(key_pair_js0.clone())
+        .unwrap()
+        .publicKey
+        .unwrap();
+    let dpk0 = DeterministicPublicKey::from(array_ref![dpk_bytes0, 0, G2_COMPRESSED_SIZE]);
+    let request = BoundedBlsSignatureRequestContextRequest {
+        issuerPublicKey: dpk0,
+        proverSecretKey: string_to_typed_bytes("WALLET_MASTER_SECRET"),
+        messageCount: 3,
+        nonce: b"dummy nonce".to_vec(),
+    };
+    let js_value = serde_wasm_bindgen::to_value(&request).unwrap();
+    let result = bounded_bls_signature_request(js_value).await;
+    assert!(result.is_ok());
+    let result: BoundedBlsSignatureRequestContextResponse = result.unwrap().try_into().unwrap();
+    let blindingFactor = result.blindingFactor;
+
+    // blind sign
+    let messages = vec![
+        string_to_typed_bytes("Message1"),
+        string_to_typed_bytes("Message2"),
+        string_to_typed_bytes("Message3"),
+    ];
+    let request = BoundedBlsSignContextRequest {
+        keyPair: serde_wasm_bindgen::from_value::<BlsKeyPair>(key_pair_js0.clone()).unwrap(),
+        messages: messages.clone(),
+        commitment: result.commitment.clone(),
+    };
+    let js_value = serde_wasm_bindgen::to_value(&request).unwrap();
+    let result = bounded_bls_sign(js_value).await;
+    assert!(result.is_ok());
+    let result: BlindSignature = result.unwrap().try_into().unwrap();
+
+    // unblind
+    let request = UnblindBoundedSignatureRequest {
+        signature: result,
+        blindingFactor,
+    };
+    let js_value = serde_wasm_bindgen::to_value(&request).unwrap();
+    let result = unblind_bounded_bls_signature(js_value).await;
+    assert!(result.is_ok());
+}
```

Comment:
```suggestion
    let signature: Signature = serde_wasm_bindgen::from_value(result.unwrap()).unwrap();

    // verify unblinded signature
    let messages_with_secret = vec![
        string_to_typed_bytes("Message1"),
        string_to_typed_bytes("Message2"),
        string_to_typed_bytes("Message3"),
        string_to_typed_bytes("WALLET_MASTER_SECRET"), // append the prover secret
    ];
    let verify_request = BlsBbsVerifyRequest {
        publicKey: dpk0,
        signature,
        messages: messages_with_secret,
    };
    let verify_request_js = serde_wasm_bindgen::to_value(&verify_request).unwrap();
    let verify_result_js = bls_verify(verify_request_js).await.unwrap();
    let verify_result =
        serde_wasm_bindgen::from_value::<BbsVerifyResponse>(verify_result_js).unwrap();
    assert!(verify_result.verified);
}
```
Added a test to assure that we can verify the unblinded signature correctly.

Reply:
Thank you for adding a test. 
I also changed from `WALLET_MASTER_SECRET` to `PROVER_SECRET`.